exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterBuilder.java

```diff
@@ -114,6 +129,51 @@ public OtlpGrpcSpanExporter build() {
         managedChannelBuilder.intercept(MetadataUtils.newAttachHeadersInterceptor(metadata));
       }
 
+      if (trustedCertificatesPem != null) {
+        // gRPC does not abstract TLS configuration so we need to check the implementation and act
+        // accordingly.
+        if (managedChannelBuilder
+            .getClass()
+            .getName()
+            .equals("io.grpc.netty.NettyChannelBuilder")) {
```

Comment:
Interesting, I was wondering why you didn't use `instanceof` and then I found that `instanceof` does load the class: https://bugs.openjdk.java.net/browse/JDK-6592631

Reply:
I also didn't realize that `instanceof` caused classload, so this is interesting!  Thanks for sharing that @Oberon00 . 
 
Is the goal here really to prevent loading one of two classes, though?  It's definitely less readable and more fragile this way, so maybe I'm missing some some other awful byproduct of the classload?

Reply:
Isn't the real reason for this to make sure we don't need to pull in these dependencies into the project, not to actually save class loading time/memory?

Reply:
I could believe that.

Reply:
Yeah user brings in the gRPC implementation they want to use (usually one of grpc-netty or grpc-netty-shaded)